pkg/diagnostics/http_monitoring.go

```diff
@@ -190,13 +235,21 @@ func (h *httpMetrics) ClientRequestCompleted(ctx context.Context, method, path,
 			diagUtils.WithTags(h.clientRoundtripLatency.Name(), appIDKey, h.appID, httpPathKey, h.convertPathToMetricLabel(path), httpMethodKey, method, httpStatusCodeKey, status),
 			h.clientRoundtripLatency.M(elapsed))
 	} else {
+		completedCountTags := diagUtils.WithTags(h.clientCompletedCount.Name(), appIDKey, h.appID, httpStatusCodeKey, status)
+		roundTripLatencyTags := diagUtils.WithTags(h.clientRoundtripLatency.Name(), appIDKey, h.appID, httpStatusCodeKey, status)
+
+		if h.pathNormalization.enabled {
+			completedCountTags = append(completedCountTags, diagUtils.WithTags(h.clientCompletedCount.Name(), httpPathKey, path, httpMethodKey, method)...)
+			roundTripLatencyTags = append(roundTripLatencyTags, diagUtils.WithTags(h.clientRoundtripLatency.Name(), httpPathKey, path, httpMethodKey, method)...)
+		}
```

Comment:
Using `append` causes re-allocations in a hoth path. Best to create 2 separate slices for the enabled and disabled cases

Reply:
Since the tags are the same as in legacy mode instead of having 2 slices I added `|| h.pathnormalization.enabled` to the if.  Do you agree with this change?

Reply:
As long as it makes sense works for me!